### ISSUE TITLE
NVDA should have 2 sets of key bindings.

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#139]
**Reported by cnorman on 7 Apr 2008 17:25 UTC**
It should be possible to set NVDA to use a "laptop" and a "desktop" keyboard layout, possibly even allowing the user to change the modifier key, possibly to the caps lock key for example

**Comment by michaelDCurran** [comment]
**Comment by pvagner on 7 Apr 2008 18:05 UTC**
In the current shapshot or even in 0.6p1 version it is possible to use capslock, insert and numpad insert as NVDA modifier keys.
There were some alternate keyboard layouts posted to the NVDA users list. They have not been well tested yet I think. so we may attach them here to make it available until something it's included in the official builds.

Also when new layouts are added we might add a combobox for it into the keyboard settings dialog.

**Comment by michaelDCurran** [comment]
**Comment by doug on 8 Apr 2008 12:16 UTC**

I have attached the latest _default_laptop.kbd I posted to the nvda list on April 2. 


Replying to [cnorman](ticket:60):
> It should be possible to set NVDA to use a "laptop" and a "desktop" keyboard layout, possibly even allowing the user to change the modifier key, possibly to the caps lock key for example

**Comment by michaelDCurran** [comment]
**Comment by doug on 30 Apr 2008 02:52 UTC**
I am Replacing _default_laptop.kbd with my latest copy which reassigns the synthSetting ring keys 
to the renamed scripts in svn-1980

**Comment by michaelDCurran** [comment]
**Comment by mdcurran on 30 Apr 2008 04:40 UTC**
Added _default_laptop.kbd to svn r1990 (Thanks Doug). Added the keyboard layout combo box to svn r1989. This really needs some testing. Especially the support for appModule-specific layouts. 
In regards to setting the modifier to the capslock, the user has been able to do this in the keyboard settings dialog for a long time.

**Comment by michaelDCurran** [comment]
**Comment by peter on 1 Aug 2010 09:50 UTC**
I tried to make this layout (see default_laptop2.kbd) more logical and easyer to remember. all often used object navigator functions are controlled by letters m,.jkl;uiop when are the modifiers ctrl+shift+nvda pressed. I also changed review cursor controls as follows. To read by lines and letters arrow keys (again in combination with ctrl+shift+nvda) are used. home and end goes to the begining and the end of current line and pageup and page down moves review cursor to first and last line of currently reviewed object. There is one irationality because you have to release ctrl and use left and right arrows to read by words. I sometimes teach blind students and they are often littlebit frustrated when they are using NVDA because using letters to read by characters, words and lines is quite unusual for many of them. so my main reason why I am trying to make laptop layout more clear and logical is to help new users (also my students :)) to quickly understand and use main principles in nvda.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 2 Aug 2010 21:13 UTC**
Please move this discussion to #798. This is a change to already implemented functionality, so should be a separate ticket.

In response to your changes, while i understand the confusion with using letters instead of arrow keys, this is contrary to NVDA's normal review cursor keyboard paradigm. That is, the review cursor has always worked with keys in a 3 by 3 grid, with top to bottom being line/word/char and left to right being previous/current/next. Whether you like this is a separate matter, but I don't think the laptop and desktop layouts should ever be so fundamentally different. Having said that, the current laptop layout definitely needs some work. We'll continue that discussion in #798.
